### PR TITLE
:book: Add breaking changes section to introduction.md

### PR DESCRIPTION
**Pull request by joekr**
**What this PR does / why we need it**:
We want a way to call out some upcoming breaking changes for new versions. This adds a note style callout in the
`ClusterAPI documentation versions` section.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Ref #8038

### CODE REVIEW

**Comment by joekr** [comment]
local book build preview

image

File: docs/book/src/introduction.md [hunk 1]
```diff
@@ -25,6 +25,14 @@ This book documents ClusterAPI v1.4. For other Cluster API versions please see t
 * [release-0-4.cluster-api.sigs.k8s.io](https://release-0-4.cluster-api.sigs.k8s.io)
 * [release-0-3.cluster-api.sigs.k8s.io](https://release-0-3.cluster-api.sigs.k8s.io)
 
+> ⚠️ Break Changes ⚠️
```
Comment (oscr) [inline docs/book/src/introduction.md:28]: Should be "Breaking Changes"

File: docs/book/src/introduction.md [hunk 2]
```diff
@@ -25,6 +25,14 @@ This book documents ClusterAPI v1.4. For other Cluster API versions please see t
...
+>
+> API Version v1alpha3 support will be removed with the upcoming release of v1.5
```
Comment (oscr) [inline docs/book/src/introduction.md:30]: Should be "API version"?

File: docs/book/src/introduction.md [hunk 3]
```diff
@@ -25,6 +25,14 @@ This book documents ClusterAPI v1.4. For other Cluster API versions please see t
...
+>
+> API Version v1alpha4 support will be removed with the upcoming release of v1.6
```
Comment (oscr) [inline docs/book/src/introduction.md:32]: Same as above

File: docs/book/src/introduction.md (hunk 2 above)
Comment (oscr) [inline docs/book/src/introduction.md:30]: Question: is it "with the" or "in the"?

File: docs/book/src/introduction.md (hunk 1 above)
Comment (joekr) [reply docs/book/src/introduction.md:28]: 🙈

File: docs/book/src/introduction.md (hunk 2 above)
Comment (joekr) [reply docs/book/src/introduction.md:30]: I think either can work here. I'll change to `in the`

**Review by oscr — APPROVED** [review]
Great work! Thank you for adding this

/lgtm

**Review by killianmuldoon — COMMENTED** [review]
Good idea!

/lgtm

File: docs/book/src/introduction.md [hunk 4]
```diff
@@ -25,6 +25,14 @@ This book documents ClusterAPI v1.4. For other Cluster API versions please see t
 * [release-0-4.cluster-api.sigs.k8s.io](https://release-0-4.cluster-api.sigs.k8s.io)
 * [release-0-3.cluster-api.sigs.k8s.io](https://release-0-3.cluster-api.sigs.k8s.io)
 
+> ⚠️ Breaking Changes ⚠️
+>
+> API version v1alpha3 support will be removed in the upcoming release of v1.5
+>
+> API version v1alpha4 support will be removed in the upcoming release of v1.6
+>
+> Review the [CONTRIBUTING support-and-guarantees](https://github.com/kubernetes-sigs/cluster-api/blob/main/CONTRIBUTING.md#support-and-guarantees) for more details
```
Comment (richardcase) [inline docs/book/src/introduction.md:34]:
```suggestion
> Review the [support-and-guarantees](./CONTRIBUTING.md#support-and-guarantees) section of the contributing guide for more details
```

**Review by richardcase — COMMENTED** [review]
I like this idea 👍

File: docs/book/src/introduction.md [hunk 5]
```diff
@@ -25,6 +25,14 @@ This book documents ClusterAPI v1.4. For other Cluster API versions please see t
 * [release-0-4.cluster-api.sigs.k8s.io](https://release-0-4.cluster-api.sigs.k8s.io)
 * [release-0-3.cluster-api.sigs.k8s.io](https://release-0-3.cluster-api.sigs.k8s.io)
 
+> ⚠️ Breaking Changes ⚠️
+>
```
[the remark this answers is not in the document]
Comment (richardcase) [reply docs/book/src/introduction.md:29]: It would bring more attention to it 👍

File: docs/book/src/introduction.md (hunk 4 above)
Comment (joekr) [reply docs/book/src/introduction.md]: updated.

File: docs/book/src/introduction.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (joekr) [reply docs/book/src/introduction.md]: moved it above getting started.

**Comment by sbueringer** [comment]
/lgtm

@joekr Just fyi, preview is also available here: https://deploy-preview-8165--kubernetes-sigs-cluster-api.netlify.app/
(link can be found in the "deploy/netlify" check)

**Pull request merged**